src/main/java/org/opentripplanner/middleware/trip_monitor/jobs/CheckMonitoredTrip.java

```diff
@@ -0,0 +1,207 @@
+        }
+    }
+
+    private void checkTripForDelays(MonitoredTrip trip, Itinerary itinerary) {
+        // TODO
+    }
+
+    private void checkTripForItineraryChanges(MonitoredTrip trip, Itinerary itinerary) {
+        // TODO
+    }
+
+    private void sendNotification() {
+        if (notifications.size() == 0) {
+            // FIXME: Change log level
+            LOG.info("No notifications queued up. Skipping notify.");
+            return;
+        }
+        OtpUser otpUser = Persistence.otpUsers.getById(trip.userId);
+        if (otpUser == null) {
+            LOG.error("Cannot find user for id {}", trip.userId);
+            // TODO: Bugsnag / delete monitored trip?
+            return;
+        }
+        String name = trip.tripName != null ? trip.tripName : "Trip for " + otpUser.email;
+        String subject = name + " Notification";
+        StringBuilder body = new StringBuilder();
+        for (TripMonitorNotification notification : notifications) {
+            body.append(notification.body);
+        }
+        // FIXME: Change log level
+        LOG.info("Sending notification '{}' to user {}", subject, trip.userId);
+        // FIXME: This needs to be an enum.
+        switch (otpUser.notificationChannel.toLowerCase()) {
+            // TODO: Use medium-specific messages (i.e., SMS body should be shorter/phone friendly)
+            case "sms":
+                NotificationUtils.sendSMS(otpUser.phoneNumber, body.toString());
+                break;
+            case "email":
+                NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            case "all":
+                NotificationUtils.sendSMS(otpUser.phoneNumber, body.toString());
+                NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void enqueueNotification(TripMonitorNotification tripMonitorNotification) {
+        notifications.add(tripMonitorNotification);
+    }
+
+    private boolean shouldSkipMonitoredTripCheck(MonitoredTrip trip) {
+        // Default to system's zone if one can't be inferred from trip destination.
+        ZoneId zoneId = ZoneId.systemDefault();
+        // TODO: Verify that this check doesn't cost anything.
+        Optional<ZoneId> zoneIdForCoordinates = getZoneIdForCoordinates(trip.to.lat, trip.to.lon);
+        if (zoneIdForCoordinates.isEmpty()) {
+            LOG.error("Could not find timezone for monitored trip: {}", trip.id);
+        } else {
+            zoneId = zoneIdForCoordinates.get();
+        }
+        // Get current time and trip time for comparison.
+        LocalDate now = LocalDate.now(zoneId);
+        LocalDate tripTime = trip.itinerary.startTime.toInstant()
+            .atZone(zoneId)
+            .toLocalDate();
```

Comment:
This seems like it will constantly check the original startTime in the itinerary and not update to a new date + time each check.

Reply:
Offset trip time with current date.